### PR TITLE
linux 18.04 base, MAAS, and gecko_t_linux_talos roles

### PR DESCRIPTION
**Pull request by aerickson**
## overview

- adds the following role_profiles
  - linux 1804 base
  - gecko_t_linux_talos
  - maas region and maas rack
- sets up test-kitchen testing for the above

---

## notes on gecko_t_linux_talos

Ports `toplevel::worker::releng::generic_worker::test::gpu` from build_puppet to ronin_puppet.

Top level definition in this repo is at `modules/roles_profiles/manifests/roles/gecko_t_linux_talos.pp`.

### SKIPPED

- graphics drivers (intel)
  - intel p580 drivers should be included
  - see https://bugzilla.mozilla.org/show_bug.cgi?id=1424465#c3 about dhouse's previous issues
- alsa sound driver
  - use pulse now?

### secrets that need setting

```
            $user = lookup('linux_vnc_user')
            $user_homedir = lookup('linux_vnc_user_homedir')
            $group = lookup('linux_vnc_group')
            $password = lookup('linux_vnc_password')
```

### how to test

see README.md in this commit

### CODE REVIEW

**Comment by davehouse** [comment]
thank you! I have not used kitchen so this is a huge help to have it ready.

**Comment by aerickson** [comment]
For sure. :) I'll update the README.md with the howto I have above.

Should I just continue adding everything that the gecko-t-linux-talos/t-linux64-ms hosts in build-puppet have?

**Comment by davehouse** [comment]
Yes, I'm good with that.

**Comment by aerickson** [comment]
Talked with Jake. This isn't 100% yet, but he thought we should land to avoid bit rot.

**Comment by aerickson** [comment]
rebased

File: modules/linux_vnc/manifests/init.pp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+class linux_vnc {
+  # include config
+  # include users::builder
+  include linux_packages::vnc_server
+  include roles_profiles::profiles::cltbld_user
```
Comment (dividehex) [inline modules/linux_vnc/manifests/init.pp:11]: do not include profiles inside component modules

File: modules/linux_vnc/manifests/init.pp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  if (lookup('cltbld_user.vnc_password') == '') {
+    fail('No VNC password set')
+  }
```
Comment (dividehex) [inline modules/linux_vnc/manifests/init.pp:15]: hiera lookups should only be done within profiles and then passed as args to the class

File: modules/linux_packages/manifests/python3_zstandard.pp [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+class linux_packages::python3_zstandard {
+    require linux_packages::py3
+
+    # for the first run, this tries to run pip because pip3 didn't exist
+    # at the start at the start of the run. ugh. puppet bug in 5.5.1.
```
Comment (dividehex) [inline modules/linux_packages/manifests/python3_zstandard.pp:9]: Is the still a bug in puppet 6?  We should be using the latest puppet.

Comment (aerickson) [reply modules/linux_packages/manifests/python3_zstandard.pp:9]: It was always working in 6. Not sure why I thought we were locked at 5 (maybe the servers I looked at were using it or ???). It was awhile ago when I wrote that.

Will change.

File: modules/linux_vnc/manifests/init.pp (hunk 1 above)
Comment (aerickson) [reply modules/linux_vnc/manifests/init.pp]: This instance has been fixed.

File: modules/linux_vnc/manifests/init.pp (hunk 2 above)
Comment (aerickson) [reply modules/linux_vnc/manifests/init.pp]: Fixed all instances.

File: modules/linux_vnc/manifests/init.pp (hunk 1 above)
Comment (aerickson) [reply modules/linux_vnc/manifests/init.pp]: All instances fixed.

**Pull request merged**